shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngine.java

```diff
@@ -214,10 +236,21 @@ private Collection<String> routeDataSources(final TableRule tableRule, final Sha
     
     private Collection<DataNode> routeTables(final TableRule tableRule, final String routedDataSource,
                                              final ShardingStrategy tableShardingStrategy, final List<ShardingConditionValue> tableShardingValues) {
-        Collection<String> availableTargetTables = tableRule.getActualTableNames(routedDataSource);
-        Collection<String> routedTables = tableShardingValues.isEmpty()
-                ? availableTargetTables
-                : tableShardingStrategy.doSharding(availableTargetTables, tableShardingValues, tableRule.getTableDataNode(), properties);
+        Collection<String> routedTables = null;
+        if (!(tableShardingStrategy instanceof HintShardingStrategy) && !tableShardingValues.isEmpty()) {
+            ShardingConditionValue shardingConditionValue = tableShardingValues.iterator().next();
+            List<ShardingConditionValue> tableHintShardingValues = getTableShardingValuesFromHint(shardingConditionValue);
+            if (!tableHintShardingValues.isEmpty()) {
+                Collection<String> availableTargetTables = tableRule.getActualTableNames(routedDataSource);
+                routedTables = new LinkedHashSet<>(tableShardingStrategy.doSharding(availableTargetTables, tableHintShardingValues, tableRule.getTableDataNode(), properties));
+            }
+        }
+        if (routedTables == null || routedTables.isEmpty()) {
+            Collection<String> availableTargetTables = tableRule.getActualTableNames(routedDataSource);
+            routedTables = new LinkedHashSet<>(tableShardingValues.isEmpty()
+                    ? availableTargetTables
+                    : tableShardingStrategy.doSharding(availableTargetTables, tableShardingValues, tableRule.getTableDataNode(), properties));
+        }
```

Comment:
There are too many nested if, could you refactor them?

Reply:
ok，me refactor the code

Reply:
refactor after code 
```
Collection<String> routedTables = null;
        if (!(tableShardingStrategy instanceof HintShardingStrategy)
                && existHintTableShardingValues()
                && !tableShardingValues.isEmpty()) {
            ShardingConditionValue shardingConditionValue = tableShardingValues.iterator().next();
            List<ShardingConditionValue> tableHintShardingValues = getTableShardingValuesFromHint(shardingConditionValue);
            Collection<String> availableTargetTables = tableRule.getActualTableNames(routedDataSource);
            routedTables = new LinkedHashSet<>(tableShardingStrategy.doSharding(availableTargetTables, tableHintShardingValues, tableRule.getTableDataNode(), properties));
        }
        if (routedTables == null) {
            Collection<String> availableTargetTables = tableRule.getActualTableNames(routedDataSource);
            routedTables = new LinkedHashSet<>(tableShardingValues.isEmpty()
                    ? availableTargetTables
                    : tableShardingStrategy.doSharding(availableTargetTables, tableShardingValues, tableRule.getTableDataNode(), properties));
        }
```

Reply:
great

Reply:
I refactor the logic,moved to create the sharding condition address,this look more concise